tests/core/gadget-update-pc/task.yaml

```diff
@@ -173,8 +177,10 @@ execute: |
         fi
 
         # prepare & install the next update
-        new_snap_declaration "$BLOB_DIR" pc_x3.snap --snap-decl-json decl-headers.json
-        new_snap_revision "$BLOB_DIR" pc_x3.snap --snap-rev-json rev-headers-3.json
+        p=$(new_snap_declaration "$BLOB_DIR" pc_x3.snap --snap-decl-json decl-headers.json)
```

Comment:
mmm if you moved the --noack implementation into the `new_snap_...` function implementations (and then just passed the options from `make_snap_installable...` to `new_snap_...` you could get rid of the extra `snap ack` we do here

Reply:
no strong opinion from me on which is better, either approach is fine to me

Reply:
That was my initial approach, until I realized that I would have then had to replicate the same `--ack` argument parsing to all functions. Without replicating the logic, `make_snap_installable()` and `make_snap_installable_with_id()` would just blindly forward all the extra arguments to `new_snap_declaration()` and `new_snap_revision()`, but I didn't like the fact that in that case `--ack` had to be passed at the *end* of the parameter list, and this the would make the implementation of the latter two functions harder (since they already forward `"$@"` to their callee).

So, I think that overall the current approach is simpler.

Reply:
thanks for explaining